pkg/osbuild/osbuild.go

```diff
@@ -51,3 +56,31 @@ func (p *Pipeline) AddStages(stages ...*Stage) {
 		p.AddStage(stage)
 	}
 }
+
+// Take some bytes and deserialize them into a Manifest; mostly used to take
+// an inspected manifest
+func NewManifestFromBytes(data []byte) (*Manifest, error) {
```

Comment:
I tried this for a real manifest but it did not go very well, it dies with `json: cannot unmarshal object into Go struct field Stage.pipelines.stages.inputs of type osbuild.Inputs` and looking at our osbuild inputs it seems we need to add quite a bit of custom unmarshaling in this area. What is the idea here? Will we expand the osbuild package so that it can unmarshal things properly back? it would be nice for tests as well to be able to transform the json back into more managable structs.